src/scenario.py

```diff
@@ -68,17 +66,17 @@ def run(self, strategy_name, options):
 
         event_steps = self.events.get_event_steps(self.start_time, self.n_intervals, self.interval)
 
-        socs = []
-        costs = []
-        prices = []
+        socs = {gcID: [] for gcID in self.constants.grid_connectors.keys()}
+        costs = {gcID: [] for gcID in self.constants.grid_connectors.keys()}
+        prices = {gcID: [] for gcID in self.constants.grid_connectors.keys()}
```

Comment:
save a shorthand to iterate over GCs: `gc_ids = self.constants.grid_connectors.keys()`
This will also come in useful later in the code.